warehouse/forklift/legacy.py

```diff
@@ -413,6 +414,40 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(_form, field):
+    declared_dynamic_fields = set(map(str.title, field.data or []))
+    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}
+    if invalid := (declared_dynamic_fields & disallowed_dynamic_fields):
+        raise wtforms.validators.ValidationError(
+            f"The following metadata field(s) cannot be marked as dynamic: {invalid!r}",
+        )
+    allowed_dynamic_field = set(DynamicFieldsEnum.enums)
+    if invalid := (declared_dynamic_fields - allowed_dynamic_field):
+        raise wtforms.validators.ValidationError(
+            f"The following values are not valid for Dynamic: {invalid!r}"
+        )
```

Comment:
I think what we actually want to do here is check if all the values in `Dynamic` are present as other metadata fields -- other verification checks will ensure that these fields are valid metadata fields for that metadata version. I'm not sure we need the enum at all.

Reply:
What I meant by this was whether this check should ensure that all the fields listed in `Dynamic` are actually provided as metadata -- e.g., if we have `Dynamic: Author` but `Author` is not provided, should this fail? I think the PEP is unclear on this.

Reply:
I think both cases allowed, e.g. if you have `Dynamic: Requires-Dist`, you may not provide any requirements, but you still have to set it as `Dynamic` or consumers can assume it's a package with no dependencies. But you may give an incomplete/insufficiently constrained list and tighten it at build time (i have been told this is important for some packages that can build for different numpy version).